MagazineLayout/LayoutCore/ModelState.swift

```diff
@@ -730,6 +736,146 @@ final class ModelState {
     }
   }
 
+  private func prepareElementLocationsForFlattenedIndices() {
```

Comment:
We need to call this anytime our section / item counts change, which is why it's called from all of those functions above ^